pkg/prometheus/promcfg_test.go

```diff
@@ -465,6 +466,256 @@ alerting:
 	}
 }
 
+func TestSettingHonorTimestamps(t *testing.T) {
```

Comment:
As this is done for both PodMonitor and ServiceMonitor I would expect both to be tested. Here we are only testing ServiceMonitor.

Reply:
Test added

Reply:
Sure, but thats outside of the scope of this PR.